documentation/docs/01-routing.md

```diff
@@ -114,28 +111,90 @@ The job of this function is to return a `{ status, headers, body }` object repre
 - `4xx` — client error
 - `5xx` — server error
 
-If the returned `body` is an object, and no `content-type` header is returned, it will automatically be turned into a JSON response. (Don't worry about `$lib`, we'll get to that [later](#modules-$lib).)
-
 > If `{fallthrough: true}` is returned SvelteKit will [fall through](#routing-advanced-fallthrough-routes) to other routes until something responds, or will respond with a generic 404.
 
-For endpoints that handle other HTTP methods, like POST, export the corresponding function:
+The returned `body` corresponds to the page's props:
+
+```svelte
+<script>
+	// populated with data from the endpoint
+	export let item;
+</script>
+
+<h1>{item.title}</h1>
+```
+
+#### POST, PUT, PATCH, DELETE
+
+Endpoints can handle any HTTP method — not just `GET` — by exporting the corresponding function:
 
 ```js
 export function post(event) {...}
+export function put(event) {...}
+export function patch(event) {...}
+export function del(event) {...} // `delete` is a reserved word
 ```
 
-Since `delete` is a reserved word in JavaScript, DELETE requests are handled with a `del` function.
+These functions can, like `get`, return a `body` that will be passed to the page as props. Whereas 4xx/5xx responses from `get` will result in an error page rendering, similar responses to non-GET requests do not, allowing you to do things like render form validation errors:
```

Comment:
I'm confused about this part of this feature. Is this for AJAX forms? Old-fashioned no-JS forms? Both?

Reply:
old-fashioned. if you hit the route with an `accept: application/json` header, it will behave like a standalone endpoint

Reply:
I think most of my questions are answered by looking at the code sample below. I don't know whether there's something I would have preferred this paragraph before the code sample to say that would have made me less temporarily confused.

But, what happens if `GET`s or `POST`s return a 200 or non-200 response that's not JSON? Does that always render an error page?

Reply:
Oh, whoops, GitHub didn't update to tell me about your comment that came in while I was writing my second one.

That might be worth mentioning in the docs here, along with the `accept: application/json` thing.

Reply:
6cb75dbd69f97942de78b57729dd7194465fb411

Reply:
Sounds good!